docs/design.md

```diff
@@ -0,0 +1,54 @@
+Design
+======
+
+Context and scope
+-----------------
+
+pod-security-admission is a [Kubernetes Admission Webhook](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/) 
+to ensure [Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/).
+
+pod-security-admission aims to be a simple [Pod Security Policy](https://kubernetes.io/docs/concepts/policy/pod-security-policy/) replacement.
+
+### Background
+
+The enforcement of security settings for Pods has previously been achieved using Pod Security Policy (PSP).
+However, it was [announced](https://github.com/kubernetes/kubernetes/pull/97171) that PSP is going to be removed in Kubernetes 1.25.
+We need technology to replace PSP.
+
+### Goals
+
+- Enforcement of Pod Security Standards
+- Easy to use without complicated settings
+- Can specify the group of Pods to which each policy is applied
+- Can customize rules to be applied for each policy 
+
+### Non-goals
+
+- Providing flexible policy engine, such as adding a new rule, customizing rules beyond Pod Security Standards
+
+Webhooks
+--------
+
+Pod Security Standards define three policy types: `Privileged`, `Baseline` and `Restricted`.
+pod-security-admission will provide ValidatingWebhook and MutatingWebhook endpoints for each of these policies.
+However, for `Priviliged`, the policy can be archived by not applying the webhook endpoint.
```

Comment:
If users don't apply any of the four webhook described below, it means to apply the `Priviliged` policy since nothing is going to restrict the user's action. Is my understanding correct?

Reply:
Yes.